clustermesh/clustermesh.go

```diff
@@ -2178,45 +1550,12 @@ func (k *K8sClusterMesh) ConnectWithHelm(ctx context.Context) error {
 	return nil
 }
 
-// Helm-based clustermesh connect/disconnect is only supported on Cilium
-// v1.14+ due to a lack of support in earlier versions for
-// autoconfigured certificates (tls.{crt,key}) for cluster
-// members when running in certgen (cronJob) PKI mode
-func (k *K8sClusterMesh) needsClassicMode(r *release.Release) (bool, error) {
```

Comment:
i _think_ it's ok to delete this logic because we already document that the latest release is only compatible with cilium 1.14 and newer here => https://github.com/cilium/cilium-cli?tab=readme-ov-file#releases